### PR TITLE
`Codegen` module

### PR DESCRIPTION
**Pull request by dmdashenkov**
In this PR we reduce code duplicates be introducing `codegen` module. The role of this module is to contain shared utilities and values for Java code generation.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov, PTAL.

File: codegen/src/main/java/io/spine/tools/CodeGeneration.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    static {
+        final CodeBlock generatedByDescription = CodeBlock.of("\"by Spine compiler\"");
+        GENERATED = AnnotationSpec.builder(Generated.class)
+                                  .addMember(GENERATED_FIELD_NAME, generatedByDescription)
+                                  .build();
+    }
+
+    /**
+     * The {@code private} constructor prevents the utility class instantiation.
```
Comment (alexander-yevsyukov) [inline codegen/src/main/java/io/spine/tools/CodeGeneration.java]: Please remove this piece:

```java
The {@code private} constructor 
```

It duplicates the code we see. Too much drama.

File: codegen/src/main/java/io/spine/tools/CodeGeneration.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+public final class CodeGeneration {
+
+    private static final AnnotationSpec GENERATED;
+
+    @SuppressWarnings("DuplicateStringLiteralInspection")
+        // Each occurrence has a different semantics.
+    private static final String GENERATED_FIELD_NAME = "value";
+
+    static {
+        final CodeBlock generatedByDescription = CodeBlock.of("\"by Spine compiler\"");
```
Comment (alexander-yevsyukov) [inline codegen/src/main/java/io/spine/tools/CodeGeneration.java]: Please have `by Spine Model Compiler`.

Comment (alexander-yevsyukov) [inline codegen/src/main/java/io/spine/tools/CodeGeneration.java]: Why cannot we inline this variable?

File: codegen/src/main/java/io/spine/tools/CodeGeneration.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import com.squareup.javapoet.CodeBlock;
+
+import javax.annotation.Generated;
+
+/**
+ * A factory for the commonly used Java source specs.
+ *
+ * @author Dmytro Dashenkov
+ */
+public final class CodeGeneration {
```
Comment (alexander-yevsyukov) [inline codegen/src/main/java/io/spine/tools/CodeGeneration.java]: Please rename to `Annotations`, at least for now.

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov, PTAL again.

**Pull request merged**